### PR TITLE
Fix theme not saving between application sessions

### PR DESCRIPTION
**Pull request by toadkarter**
**[DRAFT]**

### CODE REVIEW

File: src/application/user_settings.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        instance
+    }
+}
+
+impl UserSettings {
+    fn new() -> Self {
+        Self {
+            user_settings: HashMap::new(),
+            config_name: "config.json".to_string(),
```
Comment (Spydr06) [inline src/application/user_settings.rs:44]: Sadly, we cannot expect that the current working directory is always writable, we have to use the platform's config directory.

- **Linux** and **MacOS**: if defined, use `$XDG_CONFIG_HOME/logicrs/config.json` else `$HOME/.config/logicrs/config.json`
- **Windows**: use `%USERPROFILE%\AppData\Local\logicrs\config.json`

(environment variables need to be expanded)

Comment (toadkarter) [reply src/application/user_settings.rs:44]: Got it, thanks! Will make that tweak. Just as an FYI this PR is still draft form so not all functionality is there (but thank you for taking the time to have a look already!)

Comment (Spydr06) [reply src/application/user_settings.rs:44]: Yea, no pressure, just wanted to inform you :)

**Comment by toadkarter** [comment]
@Spydr06 The initial functionality seems to be working (still need to fix the config path like you mentioned yesterday) but I'm still struggling with getting the radio button to drop in in the appropriate state. The things that I've tried are as follows: 
- Setting the state from a callback in the ACTIONS static array, that is attached to the `change-theme` action. Unfortunately, this callback only seems to fire when you actually click the radio button, so when you open the dropdown menu it still starts in the wrong state. 
- In the ACTIONS static array in the `change-theme` action, replacing the `SystemPreference` Theme with an expression that evaluates to a Theme read from the `user-settings`. Unfortunately this is not possible because the user settings currently live inside the Application and I can't get at an instance of the application in the static array, unless I use the callback. 
- In the ACTIONS static array in the `change-theme` action, replacing the `SystemPreference` Theme with an expression that reads from `adw::StyleManager` and deduces the Theme from that. This leads to a runtime crash because it looks like ACTIONS is evaluated before `adw` is initialized.
- Setting up the `parameter_type` from `setup_gactions`. It looks like `parameter_type` is not exposed currently and I would be worried about making that mutable through a setter? 

Any thoughts about some other approaches I could try out?

**Comment by Spydr06** [comment]
> Setting up the parameter_type from setup_gactions. It looks like parameter_type is not exposed currently and I would be worried about making that mutable through a setter?

As long as it's only used for this purpose, I think this is the simplest solution

**Comment by toadkarter** [comment]
> > Setting up the parameter_type from setup_gactions. It looks like parameter_type is not exposed currently and I would be worried about making that mutable through a setter?
> 
> As long as it's only used for this purpose, I think this is the simplest solution

Cool, will give this a go after work today!

**Comment by toadkarter** [comment]
Apologies, something went wrong when rebasing my branch - I'm going to close down this pull request and just open a new one

**Pull request closed without merging**